pkg/apis/batch/validation/validation.go

```diff
@@ -665,6 +731,60 @@ func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path, opts
 	return allErrs
 }
 
+func isConditionTrue(list []batch.JobCondition, cType batch.JobConditionType) bool {
+	if condition := findConditionByType(list, cType); condition != nil {
+		return condition.Status == api.ConditionTrue
+	}
+	return false
+}
+
+func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType) *batch.JobCondition {
+	for i := range list {
+		if list[i].Type == cType {
+			return &list[i]
+		}
+	}
+	return nil
+}
+
+func validateIndexesFormat(indexesStr string, completions int) error {
```

Comment:
for api-review, these seem a bit spartan for a user. I'm not sure how to make them better. 
Either we flag this in the documentation update and mention this in more detail or we should think of more user friendly errors here.

Reply:
I would be happy to improve the messages, let me know if you have some suggestions, I think it will be helpful to make them as user-friendly as possible. Getting them right before api-review is a good thing IMO.

Reply:
maybe if you add my changes above, this becomes a bit more clear. I just think we should reference the field that had parsing issues.

Reply:
There is no need to include the field name in the messages. It is already included in the error reason.
I tested this e2e in my cluster using `kubectl edit job/<job-name> --subresource=status` and setting invalid value "abc", I get the following error:
```
# jobs.batch "indexed-job" was not valid:
# * status.completedIndexes: Invalid value: "abc": parsing error: cannot parse integer index: "abc"
```

Reply:
Ah I see.

Reply:
I browsed the codebase and could find examples both including the field and in message, and not, I'm ok to include if you think it improves something. I will apply the suggestions